### ISSUE TITLE
Integrate libuv (via luv bindings) and expose its API to Lua

### ISSUE THREAD

**Issue by Duckwhale** [evo-lua/evo-runtime#6]
Either copy/paste from evo-luvi, or maybe static FFI bindings (if that allows more control over the internals without killing usability).

**Comment by Duckwhale** [comment]
Taking a look at the luv bindings, I don't think there's any benefit to using libuv directly:

* The bindings code is generally very lightweight, so the performance impact should be minimal
  * There's maybe one extra function call, which may even be optimized away?
* luv also handles some ugly low-level details that I'd rather not touch unless I absolutely have to
* If using the FFI to access libuv directly, a thin bindings layer would also have to be implemented
  * This would effectively be no different than what the luv bindings do even if technically it may be a little more lightweight
  * It would require re-implementing the same API, and maintaining it (or rewriting some existing code)

In short, there's probably no reason to not just build in luv again (especially since it is still maintained).

If more control is needed at some point, I'm sure they'll accept a PR or (if need be) libuv APIs could be exposed from C++.

**Comment by Duckwhale** [comment]
The one question worth considering is, should the API exposed via ``uv`` or ``luv``? IIRC the luvit team favored the latter, but didn't want to change it so as to not break existing code.

I'm personally quite used to ``uv`` and I also don't know if ``luv`` really is an improvement; as far as I'm concerned how the bindings are implemented is an implementation detail and ``libuv`` is certainly too long? Since I'm too lazy to change existing code that expects the module to be called ``uv`` I'll just leave it as-is for now.
